### ISSUE TITLE
Edge crashes randomly on certain sites when they are filtered with the extension

### ISSUE THREAD

**Issue by evgeniy-ADG** [AdguardTeam/AdguardBrowserExtension#752]
With the latest Fast Ring update, Edge is experiencing random closures. This problem seems to stop when I disable (turn off ) Adguard AdBlocker. I know that’s somewhat circumstantial but best I can do at the moment. Also, I can’t absolutely confirm this but it seems like when Edge exits on a page with AdBlock turned on, the second time you call up the page it does not exit, at least most of the time.

Windows Fast Ring v1703 b16215.1000
Windows Insider v1703 b16215.
Microsoft Edge 41.16215.1000.0
Adguard AdBlocker 2.6.3.

1. They will still crash Edge with AdBlocker enabled, on opening even after being successfully viewed one or more times.
2. They do not crash Edge if “Disable filtering on this website” is selected.
3. They open successfully in Mozilla Firefox v54.0.

I have tried running AdBlocker with no other extensions installed and it seemed to make no difference.

Other extensions installed in my Edge:

Office Online v1.5.1 (turned on)
BuiltWith Technology Profiler v2.8 (turned on)
Page Analyzer Vorlon.js v1.0 (turned off)
Translator for Microsoft Edge v0.91.16.0 (turned off)

Approximately 19 out of 20 sites accessed where Edge crashes, will not crash Edge when you reopen Edge and it rebuilds it’s tabs, or the next time you access the page. But a few do.
You can start the process over again by doing Edge|Settings|Clear Browsing Data and clearing at least “Browsing history” and “Cookies and saved website data”. At that point everything Edge should think it is seeing the page for the first time.

The problem ceases if you go into Extensions|Adguard AdBlocker and set the switch to off. It is not necessary to uninstall the extension.

I figure somewhere there is a logged Edge crash message, though I have yet to locate one.

Example Sites.

<details>
https://insider.windows.com/Home/Index

https://www.visualstudio.com/vs/getting-started/

https://github.com/Microsoft/Windows-universal-samples/tree/master/Samples

https://archive.org/index.php

http://www.speedtest.net/

http://speedtest.xfinity.com/

https://developer.microsoft.com/en-us/windows/

https://www.vapordna.com
</details>

ID: 1573133

**Comment by ameshkov** [comment]
I must say that Edge in Fast Spring might be quite unstable, this bug is worth reporting to them directly.

@vozersky could you please try reproducing it on Fast Ring + AG latest?

**Comment by vmalyuta** [comment]
received the update 

Yesterday I installed Fast Ring Build 16226. I have tested a number of the sites originally reported to cause AdBlocker to exit Edge. So far the problem appears to be resolved.

Here is what is running –

Adguard AdBlocker Beta v2.6.4 (this still needs to identify as beta in Extensions)
Microsoft Edge 41.16226.1000.0
Microsoft EdgeHTML 15.16226
Windows 10 Pro Insider Preview v1703 b16226.1000
